src/golang/lib/k8s/jobs.go

```
@@ -109,6 +112,11 @@ func GetPod(ctx context.Context, name string, k8sClient *kubernetes.Clientset) (
 		return nil, err
 	}
 
+	if len(podList.Items) == 0 {
+		log.Infof("No pod has been created from job %s yet...", name)
+		return nil, ErrNoPodExists
```

Can we return the error directly? It would be good to get the stack trace.